### DISCUSSION TITLE
audio to text without loading whole wav file in memory with single result object

### DISCUSSION THREAD

**Discussion by tevslin** [deepgram/community#935] [General help]
I have large wav files altho smaller than the 2gig limit, Running in AWS I don't want to have to load the whole file into memory at once but do want it processed as one file so I get only one result object with consistent diarization. is there a way to accomplish this with the python sdk>

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment]
Hi @tevslin 

The best way would be to do this asynchronously.
https://github.com/deepgram/deepgram-python-sdk/blob/main/deepgram/clients/listen/v1/rest/client.py#L43

**Comment by tevslin** [comment]
David, thanks.

 

But this with a URL. Is there a way I can do this with a file which is local to my ECS server?

**Comment by davidvonthenen** [comment]
Hi @tevslin 

The only way to do that would be to land a process to call either of the functions below to "upload" the file to Deepgram:
[transcribe_file](https://github.com/deepgram/deepgram-python-sdk/blob/main/deepgram/clients/listen/v1/rest/client.py#L194)
OR
async using [transcribe_file_callback](https://github.com/deepgram/deepgram-python-sdk/blob/main/deepgram/clients/listen/v1/rest/client.py#L276)

If you use the suggested method in the previous post, it requires that the file be accessible to the internet as public or configured with a key, but this will trigger Deepgram to move the audio (aka download file) from the Deepgram side. No external process needed.
[transcribe_url](https://github.com/deepgram/deepgram-python-sdk/blob/main/deepgram/clients/listen/v1/rest/client.py#L43)
OR
async using [transcribe_url_callback](https://github.com/deepgram/deepgram-python-sdk/blob/main/deepgram/clients/listen/v1/rest/client.py#L123)

that is why it takes a URL as one of the inputs.
